src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -6,133 +6,179 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectorRef, ElementRef, Inject} from '@angular/core';
 import {SpecificEventListener, EventType} from '@material/base';
 import {MDCSliderAdapter, Thumb, TickMark} from '@material/slider';
+import {MatSliderThumb, MAT_SLIDER} from './slider-thumb';
 
+/**
+ * This is a dummy interface that just contains the properties and methods of MatSlider that are
+ * used by SliderAdapter. Rather than directly referencing MatSlider, we use this interface when
+ * to avoid a circular dependency between MatSlider and SliderAdapter.
+ */
+interface MatSlider {
+  _cdr: ChangeDetectorRef;
+  min: number;
+  max: number;
+  disabled: boolean;
+  _elementRef: ElementRef<HTMLElement>;
+  _trackActive: ElementRef<HTMLElement>;
+  _initialized: boolean;
+  _tickMarks: TickMark[];
+  _document: Document;
+  _window: Window;
+  displayWith: ((value: number) => string) | null;
+  _getInput: (thumb: Thumb) => MatSliderThumb;
+  _getKnobElement: (thumb: Thumb) => HTMLElement;
+  _getThumbElement: (thumb: Thumb) => HTMLElement;
+  _getInputElement: (thumb: Thumb) => HTMLInputElement;
+  _setValue: (value: number, thumb: Thumb) => void;
+  _setValueIndicatorText: (value: number, thumb: Thumb) => void;
```

Comment:
This is a lot of API surface. When we looked at this, I thought it was only 5 members that needed to be in the interface?

Reply:
That was the case for the MatSliderThumb, but we also need to do this for the SliderAdapter

Reply:
IIRC from my discussion with Wagner, this type is exposed through a public method/property somewhere, so it needs to at least have all of the public APIs from mat slider.

I would honestly rather just put this all in the same file though. Yes big files are a pain to edit, but editors these days all have fancy folding features so you can just collapse down the irrelevant stuff

Reply:
Or just add this to the list of allowed circular dependencies, if that's something we can do

Reply:
Yeah, when I suggested the interface I had assumed that it would only need a few methods. Manually maintaining a clone of the API surface does seem like it's worse than having one big file.